fdbclient/ClientLogEvents.h

```diff
@@ -22,18 +22,74 @@
 #ifndef FDBCLIENT_CLIENTLOGEVENTS_H
 #define FDBCLIENT_CLIENTLOGEVENTS_H
 
+#include "fdbclient/FDBTypes.h"
+#include "fdbclient/JSONDoc.h"
+
+#include <algorithm>
+
+struct RequestStats {
+	RequestStats() : nextReadId(0) {}
+
+	struct ReadStats {
+		ReadStats(int readId, int bytesFetched, int keysFetched, Key beginKey, Key endKey, NetworkAddress storageContacted, double latency) :
+			readId(readId), bytesFetched(bytesFetched), keysFetched(keysFetched), beginKey(beginKey), endKey(endKey), storageContacted(storageContacted), latency(latency) {}
+
+		int readId;
+		int bytesFetched;
+		int keysFetched;
+		Key beginKey;
+		Key endKey;
+		NetworkAddress storageContacted;
+		double latency;
+	};
+
+	std::vector<ReadStats> reads;
+	std::vector<NetworkAddress> proxies;
+
+	int getNextReadId() { return nextReadId++; };
+
+	json_spirit::Object getJson() const {
+		json_spirit::Array proxiesContacted;
+		for (const auto &p : proxies) {
+			proxiesContacted.push_back(json_spirit::Value(p.toString()));
+		}
+		json_spirit::Array readStatsList;
+		for (const auto &r : reads) {
+			json_spirit::Object readStats;
+			readStats.push_back(json_spirit::Pair("readId", json_spirit::Value(static_cast<uint64_t>(r.readId))));
```

Comment:
Why not use `int64_t` in the original type above itself?

Reply:
Thanks, I just switched to doing that instead.